### PR TITLE
Migrate `fidesui` to `fides/clients`

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-2067](https://ethyca.atlassian.net/browse/PROD-2067)

### Description Of Changes

- Include as part of the clients monorepo using turbo
- Replace imports in Privacy Center and Admin UI with this new package version
- Build/Bundle as part of those project using NextJS `transpileModules`
- Include all components as one flat package instead of splitting between react and components
- New README explaining the migration and package

### Code Changes
- FidesUI is now part of the turbo monorepo as an [Internal Package](https://github.com/vercel/turbo/blob/v1.9.3/docs/pages/repo/docs/handbook/sharing-code/internal-packages.mdx)
- Fides UI now gets bundled by NextJS using the [next-transpile-modules](https://www.npmjs.com/package/next-transpile-modules) npm package
- For now does not have a Storybook implementation as we investigate and determine the future of this project.
- In light of all of the above, Lerna is no longer needed or included.
- Sub-directories are now all collapsed to a single index
- search/replace any instances of the following with `fidesui`
  - `@fidesui/react`
  - `@fidesui/components`
  - `@chakra-ui/react`
  - `@chakra-ui/icons`
- rename `FidesProvider` to `FidesUIProvider` for clarity
- New README.md file for FidesUI explaining the migration and current usage

### Steps to Confirm

* Run cypress tests for `admin-ui` and `privacy-center`
* General smoke testing and regression testing.

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

[PROD-2067]: https://ethyca.atlassian.net/browse/PROD-2067?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/package.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "workspaces": [
     "admin-ui",
     "fides-js",
-    "privacy-center"
+    "privacy-center",
+    "fidesui"
```
Comment (NevilleS) [inline clients/package.json:6]: nit: When in doubt, I always prefer an alphabetic sort:
```suggestion
    "fidesui",
    "privacy-center"
```

(NOTE: This nitpick is really unnecessary but I'm leaving it hopefully so you feel _comforted_ knowing that I am looking at every file!)

**Review by NevilleS — APPROVED** [review]
Nice! This looks solid to me - a few minor questions/nits along the way

File: clients/fidesui/src/FidesUIProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {
+  ChakraProvider as BaseChakraProvider,
+  ChakraProviderProps,
+} from "@chakra-ui/react";
+import { theme as defaultTheme } from "./FidesUITheme";
+
+export const FidesUIProvider = ({
```
[the remark this answers is not in the document]
Comment (NevilleS) [reply clients/fidesui/src/FidesUIProvider.tsx:7]: Nevermind I got this backwards 🙃

File: clients/fidesui/package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "@chakra-ui/table": "^1.3.4",
+    "@chakra-ui/tabs": "^1.6.5",
+    "@chakra-ui/tag": "^1.2.5",
+    "@chakra-ui/textarea": "^1.2.6",
+    "@chakra-ui/theme": "^1.13.2",
+    "@chakra-ui/toast": "^1.5.4",
+    "@chakra-ui/tooltip": "^1.4.6",
+    "@chakra-ui/transition": "^1.4.5",
+    "@chakra-ui/utils": "^1.10.2",
+    "@chakra-ui/visually-hidden": "^1.1.4",
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/fidesui/package.json:54]: I wondered about that, wasn't sure if we were intentionally leaving it like this for finer version control. I would definitely prefer to simplify it.

File: clients/fidesui/package.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    "typescript": "^4.9.5"
+  },
+  "peerDependencies": {
+    "@chakra-ui/system": "^1.10.3",
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/fidesui/package.json:58]: I don't see a reason for it, but just moved it as is from the original. Now that it's an internal package, we probably don't need to specify any peer deps.

File: clients/package.json (hunk 1 above)
Comment (gilluminate) [reply clients/package.json:6]: good call. And keep the nitpicks coming. I appreciate them!

File: clients/fidesui/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Fides UI
```
Comment (lucanovera) [inline clients/fidesui/README.md:1]: Thanks for the readme notes

File: clients/privacy-center/pages/_app.tsx [hunk 6]
```diff
@@ -3,13 +3,6 @@ import { useMemo } from "react";
 import { ErrorBoundary } from "react-error-boundary";
 import { Provider } from "react-redux";
 import { PersistGate } from "redux-persist/integration/react";
-/*
```
Comment (lucanovera) [inline clients/privacy-center/pages/_app.tsx:6]: Nice one fixing an bug from almost 2 years ago!

**Review by lucanovera — COMMENTED** [review]
Great work! Didn't found any issues in the code or smoke testing the UI. Approved. 💯

**Review by lucanovera — APPROVED** [review]

**Comment by eastandwestwind** [comment]
Starting review...

File: clients/fidesui/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+This package is a result of migrating the [ethyca/fidesui](https://github.com/ethyca/fidesui) standalone repo into this monorepo. It includes components that are custom built for FidesUI. They may be built from scratch, or use a variety of Chakra components to create something new.
```
Comment (eastandwestwind) [inline clients/fidesui/README.md:3]: 🎆

File: clients/fidesui/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+A few of the main differences between the standalone and this are:
+
+- This is now part of the turbo monorepo when imported as an [Internal Package](https://github.com/vercel/turbo/blob/v1.9.3/docs/pages/repo/docs/handbook/sharing-code/internal-packages.mdx)
+- Fides UI now gets bundled by NextJS using the [next-transpile-modules](https://www.npmjs.com/package/next-transpile-modules) npm package
+- For now does not have a Storybook implementation as we investigate and determine the future of this project.
+- In light of all of the above, Lerna is no longer needed or included.
+
+## What should go in here
+
+In general, components that need to be shared across Fides applications should go in here in order to ensure a consistent UI. This may involve needing to import types specific to the Fides API as needed.
```
Comment (eastandwestwind) [inline clients/fidesui/README.md:16]: Can you write something briefly re: how we dev against fidesui? E.g. when we make a change here can we test directly in `clients/fidesui` or are we limited to testing full-stack using admin-ui?

**Review by eastandwestwind — CHANGES_REQUESTED** [review]
This is exciting! Nice work overall, just looking for a small quality-of-life update, below.

File: clients/fidesui/README.md (hunk 8 above)
Comment (gilluminate) [reply clients/fidesui/README.md:16]: great question and suggestion. I'm about to add linting to it, as well as type checking. Testing has always been in the admin-ui and fides-ui projects, I'll pull over some of the language from the original repo about that here as well.

File: clients/fidesui/README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Some things that may not belong in this repo ([original discussion](https://github.com/ethyca/fidesui/pull/27#discussion_r1010820430)):
+
+- Routing/pages. (Maybe helpers, but not the definitions of specific routes.)
+- Mutation types. These usually involve a sequence calls that require UI-specific handling. (Maybe basic CRUD could be shared, at most).
+
+## How to use
+
+This package is included in both the `admin-ui` and `privacy-center` NextJS applications as a turbo Internal Package called `fidesui` and will automatically get bundled in those projects by NextJS using `transpileModules`. This means that you can import components from this package directly in your NextJS application and there is no reason to prebuild these components. When making changes or creating new components here it's important to test them in the consuming applications to ensure that they work as expected.
```
Comment (eastandwestwind) [inline clients/fidesui/README.md:25]: ty!!

**Review by eastandwestwind — APPROVED** [review]

**Pull request merged**